interfaces/core.go

```diff
@@ -113,6 +113,13 @@ type Info struct {
 	Slots   []*snap.SlotInfo
 }
 
+type ConnectionInfo struct {
```

Comment:
```
ConnectionInfo describes a connection between a plug and a slot for the user
```
Question: do you intend to serialize this?

Reply:
Not directly, I envision an intermediate connectioInfoJSON struct or similiar created out of this data.

Reply:
Ah, I'm sorry, I missed the point. Ok added with small adjustment. Not sure if the name is good for what it is, as you pointed out in later comment.